aws/iam.go

```diff
@@ -0,0 +1,75 @@
+package aws
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"log"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+)
+
+type iamClient struct {
```

Comment:
See comment on the other files how I'd prefer to see this refactored implementing a CertificateProvider